### PR TITLE
ISSUE-42: Properly nest login fields.

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Fixes fieldgroup nesting issue on user login page.
- #42 

# Needed By (Date)
- ???

# Urgency
- Low, fields work but are not pretty. 

# Steps to Test

1. Check out this branch on a site that can authenticate via SAML & Local
2. Go to `/user` and review form for proper field display
3. Log in with SAML
4. Log in with local Drupal user.

# Associated Issues and/or People
-  #42 
- @wilsonww

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: stanford_ssp.module [hunk 1]
```diff
@@ -675,7 +677,17 @@ function stanford_ssp_form_user_login_alter(array &$form, array &$form_state) {
       '#type' => 'fieldset',
       '#title' => "Drupal Login",
       '#weight' => 0,
+      '#collapsible' => TRUE,
+      '#collapsed' => FALSE,
+      '#tree' => FALSE, // Ensure that the name and pass fields are not nested on validate and submit.
```
[the remark this answers is not in the document]
Comment (sherakama) [reply stanford_ssp.module]: For the sake of the comment clarity can we ignore this one?

[the remark this answers is not in the document]
Comment (sherakama) [reply stanford_ssp.module]: For the sake of the comment clarity can we ignore this one?

**Comment by wilsonww** [comment]
Tested on my sites and it works as intended; proper display fixed; logging in still works for both local and SAML.

screen shot 2018-07-19 at 11 06 55 am

**Comment by sherakama** [comment]
Woot. Thanks @wilsonww

File: stanford_ssp.module (hunk 1 above)
[the remark this answers is not in the document]
Comment (jbickar) [reply stanford_ssp.module]: Yes

[the remark this answers is not in the document]
Comment (jbickar) [reply stanford_ssp.module]: And yes.

**Comment by jbickar** [comment]
@wilsonww please pull the latest and review the placement and functionality of the "Log in" submit button.

@sherakama I changed the Local Drupal Login fieldset to collapsed by default. OK?

File: stanford_ssp.module [hunk 2]
```diff
@@ -675,7 +677,19 @@ function stanford_ssp_form_user_login_alter(array &$form, array &$form_state) {
       '#type' => 'fieldset',
       '#title' => "Drupal Login",
       '#weight' => 0,
+      '#collapsible' => TRUE,
+      '#collapsed' => TRUE,
```
Comment (sherakama) [inline stanford_ssp.module]: I left this defaulted to open as it causes unnecessary complications with our behat tests when `@javascript` is used.

Comment (jbickar) [reply stanford_ssp.module]: Hm. We had a custom step definition for that with WMD.

User experience vs. automated testing. Who will win?

Comment (sherakama) [reply stanford_ssp.module]: I don't have a hard line on this. I also don't think it is much of an addition to the user experience by hiding a login form. I think there is enough incentive to use the sunet option.

Comment (jbickar) [reply stanford_ssp.module]: I'm on the fence. I think it reduces the user temptation to enter their SUNet username/password in the Drupal login form fields. On the other hand, we guide authentication towards SSO in many other ways (e.g., global header, first sidebar, on `403`), such that an end user should not have to encounter this form.

Let me test with Behat with this change and see if it works.

Comment (jbickar) [reply stanford_ssp.module]: Hm, my Lando Behat is not behaving

**Comment by wilsonww** [comment]
Pulled latest and confirming Login button is now within the fieldset and the field is collapsed. For the 20+ sites we run, we don't use local logins frequently, as I've only had one instance where I un-collapsed that section with JS injector

File: stanford_ssp.module (hunk 2 above)
Comment (jbickar) [reply stanford_ssp.module]: OK, Behat fails to log in with this collapsed by default. I'm going to punt on that and change `#collapsed` back to `FALSE`

**Pull request merged**